### PR TITLE
Add possibility to build debug dali using build.sh

### PR DESCRIPTION
**Pull request by szalpal**
Signed-off-by: Michał Szołucha <mszolucha@nvidia.com>

### CODE REVIEW

File: docker/build.sh [hunk 1]
```diff
@@ -1,12 +1,13 @@
 #!/bin/bash -xe
 #########Set Me###############
-export PYVER=${PYVER:-2.7}
+export PYVER=${PYVER:-3.5}
```
Comment (JanuszL) [inline docker/build.sh]: Not needed.

File: docker/build.sh [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 export PYV=${PYVER/./}
 export CUDA_VERSION=${CUDA_VERSION:-10}
 export NVIDIA_BUILD_ID=${NVIDIA_BUILD_ID:-12345}
 export CREATE_WHL=${CREATE_WHL:-YES}
 export CREATE_RUNNER=${CREATE_RUNNER:-NO}
 export DALI_BUILD_FLAVOR=${DALI_BUILD_FLAVOR}
+export BUILD_TYPE=${BUILD_TYPE:-Debug}
```
Comment (JanuszL) [inline docker/build.sh]: I this Debug should not be a default.
Also you can list possible values in comment.
+Update the readme that new option is available.

File: docker/build.sh (hunk 1 above)
Comment (szalpal) [reply docker/build.sh]: Why should we keep old version as default?

Comment (JanuszL) [reply docker/build.sh]: Sounds fair

File: docker/build.sh (hunk 2 above)
Comment (jantonguirao) [inline docker/build.sh]:
```suggestion
export CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE:-Debug}
```
maybe?

Comment (szalpal) [inline docker/build.sh]:
```suggestion
export BUILD_TYPE=${BUILD_TYPE:-Release}
```

Comment (szalpal) [reply docker/build.sh]: I decided to keep it as BUILD_TYPE, since it's more generic (build.sh arg doesn't have to assume, that cmake is used in Dali)

Comment (szalpal) [reply docker/build.sh]: Done

**Comment by szalpal** [comment]
!build

File: CMakeLists.txt [hunk 3]
```diff
@@ -53,6 +53,7 @@ if (NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE "Release" CACHE STRING
     "Build type from [Debug, Release]. For perf testing, build Release" FORCE)
```
Comment (klecki) [inline CMakeLists.txt]: This is outdated, now it's:
```suggestion
    "Build type from [Debug, DevDebug, Release, RelWithDebInfo]. For perf testing, build Release" FORCE)
```

File: README.rst [hunk 4]
```diff
@@ -302,6 +302,7 @@ Building DALI using Clang (experimental):
 -  ``BUILD_NVOF`` - build with ``NVIDIA OPTICAL FLOW SDK`` support (default: ON)
 -  ``BUILD_NVDEC`` - build with ``NVIDIA NVDEC`` support (default: ON)
 -  ``BUILD_NVML`` - build with ``NVIDIA Management Library`` (``NVML``) support (default: ON)
+-  ``BUILD_TYPE`` - select build configuration. Supported configurations: Debug, Release (default: Release)
```
Comment (klecki) [inline README.rst]: This is wrong: It's CMAKE_BUILD_TYPE, and it's regular CMake thing.
Here we list mostly options and custom variables we use in CMake.
Also we have 4 configurations: Debug, DevDebug, Release, RelWithDebInfo

File: CMakeLists.txt (hunk 3 above)
Comment (JanuszL) [reply CMakeLists.txt]: True, it can be just called `CMAKE_BUILD_TYPE`.

File: README.rst (hunk 4 above)
Comment (szalpal) [inline README.rst]:
```suggestion

```

**Pull request merged**